field/numeric_test.go

```diff
@@ -37,14 +37,14 @@ func TestNumericField(t *testing.T) {
 	require.Equal(t, 9876, numeric.Value())
 
 	numeric = NewNumeric(spec)
-	numeric.SetData(NewNumericValue(9876))
+	numeric.Marshal(NewNumericValue(9876))
 	packed, err = numeric.Pack()
 	require.NoError(t, err)
 	require.Equal(t, "      9876", string(packed))
 
 	numeric = NewNumeric(spec)
 	data := NewNumericValue(0)
-	numeric.SetData(data)
+	numeric.Marshal(data)
```

Comment:
This test will fail after we remove `data` from the `Numeric` field. The behavior of `SetData` is slightly different from `Marshal`. Just test that `Marshal` sets value and `Unmarshal` sets value back to the arg.